src/wallet/wallet.cpp

```diff
@@ -1586,7 +1586,11 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
     }
     double progress_current = progress_begin;
     while (block_height && !fAbortRescan && !chain().shutdownRequested()) {
-        m_scanning_progress = (progress_current - progress_begin) / (progress_end - progress_begin);
+        if (progress_end - progress_begin > 0.0) {
```

Comment:
nit: `progress_end > progress_begin`

Reply:
Thanks a lot for reviewing!

```
if (progress_end - progress_begin > 0.0) {
```

… is meant to be "visually consistent" with the already existing L1590 …

```
if (*block_height % 100 == 0 && progress_end - progress_begin > 0.0) {
```

Does that make sense? :)

Reply:
Yeah, it's a subjective point, but clearer IMO.